src/openloco/windows/industry_list.cpp

```diff
@@ -1,16 +1,1323 @@
+                            productionTransported += industry->produced_cargo_transported[1];
+                            productionTransported = productionTransported >> 1;
+                        }
+                    }
+
+                    auto args = FormatArguments();
+                    args.push<uint16_t>(productionTransported);
+
+                    gfx::draw_string_494BBF(*dpi, 440, yPos, 138, colour::black, string_ids::production_transported_percent, &args);
+                }
+                yPos += common::rowHeight;
+            }
+        }
+
+        // 0x00458113
+        static ui::cursor_id cursor(window* self, int16_t widgetIdx, int16_t xPos, int16_t yPos, ui::cursor_id fallback)
+        {
+            if (widgetIdx != widx::scrollview)
+                return fallback;
+
+            uint16_t currentIndex = yPos / common::rowHeight;
+            if (currentIndex < self->var_83C && self->row_info[currentIndex] != -1)
+                return cursor_id::hand_pointer;
+
+            return fallback;
+        }
+
+        // 0x004580DE
+        static void event_08(window* self)
+        {
+            self->flags |= window_flags::flag_14;
+        }
+
+        // 0x004580E6
+        static void event_09(window* self)
+        {
+            if ((self->flags & window_flags::flag_14) == 0)
+                return;
+
+            if (self->row_hover == -1)
+                return;
+
+            self->row_hover = -1;
+            self->invalidate();
+        }
+
+        // 0x00457FCA
+        static void tabReset(window* self)
+        {
+            self->min_width = common::minDimensions.width;
+            self->min_height = common::minDimensions.height;
+            self->max_width = common::maxDimensions.width;
+            self->max_height = common::maxDimensions.height;
+            self->var_83C = 0;
+            self->row_hover = -1;
+            common::refreshIndustryList(self);
+        }
+
+        static void init_events()
+        {
+            events.draw = draw;
+            events.cursor = cursor;
+            events.draw_scroll = draw_scroll;
+            events.event_08 = event_08;
+            events.event_09 = event_09;
+            events.get_scroll_size = get_scroll_size;
+            events.on_mouse_up = on_mouse_up;
+            events.on_update = on_update;
+            events.scroll_mouse_down = on_scroll_mouse_down;
+            events.scroll_mouse_over = on_scroll_mouse_over;
+            events.prepare_draw = prepare_draw;
+            events.tooltip = tooltip;
+        }
+    }
+
     // 0x004577FF
     window* open()
     {
-        registers regs;
-        call(0x004577FF, regs);
-        return (window*)regs.esi;
+        auto window = WindowManager::bringToFront(WindowType::industryList, 0);
+        if (window != nullptr)
+        {
+            window->call_on_mouse_up(4);
+        }
+        else
+        {
+            // 0x00457878
+            window = WindowManager::createWindow(
```

Comment:
This doesn't look right. There was code that computed the size based on the ui_width